<!-- sct-test-summary -->
## :x: Test Summary: FAILED

### :white_check_mark: Precommit: PASSED

| Total | Passed | Failed | Skipped |
|-------|--------|--------|---------|
| 28 | 16 | 0 | 12 |


### :x: Tests: FAILED

| Total | Passed | Failed | Errors | Skipped |
|-------|--------|--------|--------|---------|
| 4369 | 4315 | 2 | 0 | 52 |

### Failed Tests

#### `integration.test_base_version.test_2024_2_ubuntu`
**Type:** FAILURE
**Message:** `AssertionError: assert {'2024.1', '6.0'} == {'2024.1', '2024.2', '6.0'}
  
  Extra items in the right set:
  '2024.2'
  
  Full diff:
    {
        '2024.1',
  -     '2024.2',
        '6.0',
    }`
<details><summary>Traceback</summary>

```
def test_2024_2_ubuntu():
        """Test that development branch of 2024.2 on ubuntu is returned correct versions"""
        scylla_repo = url_base + "-enterprise/enterprise-2024.2/deb/unified/latest/scylladb-2024.2/scylla.list"
        linux_distro = "ubuntu-focal"
        version_list = general_test(scylla_repo, linux_distro)
>       assert set(version_list) == {"6.0", "2024.1", "2024.2"}
E       AssertionError: assert {'2024.1', '6.0'} == {'2024.1', '2024.2', '6.0'}
E         
E         Extra items in the right set:
E         '2024.2'
E         
E         Full diff:
E           {
E               '2024.1',
E         -     '2024.2',
E               '6.0',
E           }

unit_tests/integration/test_base_version.py:97: AssertionError
```
</details>

#### `integration.test_vector_store.test_vector_search@docker_heavy`
**Type:** FAILURE
**Message:** `AssertionError: Self-search distance too high for API search: 0.16892582. Expected near 0.
assert 0.16892582 < 0.001`
<details><summary>Traceback</summary>

```
docker_scylla = <sdcm.utils.docker_remote.RemoteDocker object at 0x7932ee5c9d10>
docker_vector_store = <sdcm.cluster_docker.VectorStoreSetDocker object at 0x7932ee5ca710>
params = SCTConfiguration(perf_simple_query_extra_command=None, force_run_iotune=None, data_volume_disk_throughput=None, multi_...d_regions=['eu-west-1', 'eu-west-2', 'eu-west-3', 'us-west-2', 'us-east-1', 'us-east-2', 'eu-north-1', 'eu-central-1'])

    def test_vector_search(docker_scylla, docker_vector_store, params):  # noqa: PLR0914
        """Test of vector search"""
        db_cluster, vs_cluster = docker_scylla.parent_cluster, docker_vector_store
    
        create_vector_table(db_cluster)
        test_vectors = insert_test_vectors(db_cluster)
        vector_client = vs_cluster.nodes[0].get_vector_store_api_client()
        wait_for_vector_indexing(vector_client)
    
        query_vector = test_vectors[0]["vector"]
    
        # vector search via API
        search_results = vector_client.ann_search(
            keyspace="vector_test", index="embeddings_vector_idx", vector=query_vector, limit=10
        )
        primary_keys = search_results.get("primary_keys", {}).get("id", [])
        assert len(primary_keys) > 0, "No search results returned for API search"
        assert "distances" in search_results, "No distances in API search results"
        closest_distance = search_results["distances"][0]
>       assert closest_distance < 0.001, (
            f"Self-search distance too high for API search: {closest_distance}. Expected near 0."
        )
E       AssertionError: Self-search distance too high for API search: 0.16892582. Expected near 0.
E       assert 0.16892582 < 0.001

unit_tests/integration/test_vector_store.py:138: AssertionError
```
</details>

[Full build log](https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-cluster-tests/job/PR-15518/3/)

--- a/sdcm/sct_config.py
+++ b/sdcm/sct_config.py
@@ -3151,6 +3151,9 @@ class SCTConfiguration(BaseModel):
         if self.get("cluster_backend") == "azure":
             # for Azure need to shorten it more due longer region names
             prefix_max_len -= 2
+        if self.get("cluster_backend") == "oci":
+            # OCI node names include region (up to 14 chars) and must fit X.509 CN 64-char limit
+            prefix_max_len -= 10
         if (self.get("simulated_regions") or 0) > 1:
             # another shortening for simulated regions due added simulated dc suffix
             prefix_max_len -= 3

--- a/unit_tests/integration/test_config.py
+++ b/unit_tests/integration/test_config.py
@@ -75,6 +75,24 @@ def test_10_longevity(monkeypatch):
 
 
 @pytest.mark.integration
+def test_10_oci_user_prefix_fits_x509_cn_limit(monkeypatch):
+    """OCI node names include region; user_prefix must be short enough for the full name to fit X.509 CN 64-char limit."""
+    user_prefix = "longevity-5gb-1h-ToggleAuditRulesNemesisSyslog"
+    monkeypatch.setenv("SCT_CLUSTER_BACKEND", "oci")
+    monkeypatch.setenv("SCT_CONFIG_FILES", "unit_tests/test_configs/minimal_test_case.yaml")
+    monkeypatch.setenv("SCT_USER_PREFIX", user_prefix)
+    monkeypatch.setenv("SCT_OCI_IMAGE_DB", "ocid1.image.oc1.iad.fake")
+    monkeypatch.setenv("SCT_OCI_IMAGE_USERNAME", "ubuntu")
+    monkeypatch.setenv("SCT_OCI_INSTANCE_TYPE_DB", "VM.DenseIO.E5.Flex:8")
+    monkeypatch.setenv("SCT_OCI_INSTANCE_TYPE_LOADER", "VM.Standard.E4.Flex:2")
+    monkeypatch.setenv("SCT_OCI_INSTANCE_TYPE_MONITOR", "VM.Standard.E4.Flex:2")
+    monkeypatch.setenv("SCT_OCI_REGION_NAME", "eu-frankfurt-1")
+    conf = sct_config.SCTConfiguration()
+    conf.verify_configuration()
+    assert conf.user_prefix == user_prefix[:25]  # 35 - 10
+
+
+@pytest.mark.integration
 def test_10_mananger_regression(monkeypatch):
     monkeypatch.setenv("SCT_CLUSTER_BACKEND", "aws")
     monkeypatch.setenv("SCT_AMI_ID_DB_SCYLLA", "ami-dummy ami-dummy2")

